### PR TITLE
Update tooling.md

### PR DESCRIPTION
**Pull request by flip111**
https://github.com/rust-embedded/book/issues/134

### CODE REVIEW

**Comment by flip111** [comment]
https://github.com/rust-embedded/book/issues/143

File: src/intro/tooling.md [hunk 1]
```diff
@@ -15,15 +15,13 @@ tested.
...
-  have neither installed then don't worry about installing either.
+- [`cargo-generate`](https://github.com/ashleygwilliams/cargo-generate) and `git`.
+These tools are optional but will make it easier to follow along with the book.
 
 The text below explains why we are using these tools. Installation instructions
 can be found on the next page.
 
-## `cargo-generate` OR `git`
-
+## Using a template for the first project ##
```
Comment (japaric) [inline src/intro/tooling.md]: All headings are named after tools. With this change that would no longer be the case. I would prefer to keep the symmetry.

File: src/intro/tooling.md [hunk 2]
```diff
@@ -15,15 +15,13 @@ tested.
 - OpenOCD >=0.8. Tested versions: v0.9.0 and v0.10.0
 - GDB with ARM support. Version 7.12 or newer highly recommended. Tested
   versions: 7.10, 7.11, 7.12 and 8.1
-- [OPTIONAL] `git` OR
-  [`cargo-generate`](https://github.com/ashleygwilliams/cargo-generate). If you
-  have neither installed then don't worry about installing either.
+- [`cargo-generate`](https://github.com/ashleygwilliams/cargo-generate) and `git`.
```
Comment (japaric) [inline src/intro/tooling.md]: This seems fine to me

File: src/intro/tooling.md (hunk 1 above)
Comment (flip111) [reply src/intro/tooling.md]: What about `cargo-generate and other tools` ?

File: src/intro/tooling.md (hunk 2 above)
Comment (japaric) [reply src/intro/tooling.md]: Actually, this should be "(..) or `git`". You don't need *both* tools; one of them suffices.

File: src/intro/tooling.md (hunk 1 above)
Comment (flip111) [reply src/intro/tooling.md]: By the way, maybe write something about rustc too ..

Comment (japaric) [reply src/intro/tooling.md]: The current text matches what's listed in the bullet list. The "other tools" in your proposal makes it unclear whether "other tools" means just git or all / some of the other tools in the bullet list.

Personally, I think the current text is fine.

File: src/intro/tooling.md (hunk 2 above)
Comment (flip111) [reply src/intro/tooling.md]: Later on in the book `cargo generate --git` command is used. We can put `or` here when comes with it's own git api and doesn't rely on standalone git. I assumed it relies on standalone git.

Comment (flip111) [reply src/intro/tooling.md]: From this wording i conclude that standalone git is not required https://github.com/rust-lang/git2-rs#version-of-libgit2 (dependency of cargo generate)

File: src/intro/tooling.md (hunk 1 above)
Comment (japaric) [reply src/intro/tooling.md]: > By the way, maybe write something about rustc too ..

Rust developers (the target audience of this book) are familiar with rustc / Cargo; I don't think it's necessary to add a section for them.

File: src/intro/tooling.md (hunk 2 above)
Comment (japaric) [reply src/intro/tooling.md]: `cargo-generate` doesn't shell out to `git`; it uses libgit.

File: src/intro/tooling.md (hunk 1 above)
Comment (flip111) [reply src/intro/tooling.md]: i just find it strange that the choice now is between `cargo generate` and `git` both tools need to be downloaded and installed, while the other tools "curl, wget, or your web browser." are likely to be already on a system. But i'm tired to bike shed over this, so you can get the final call, i'll leave it as it is or make it whatever.

**Comment by japaric** [comment]
@flip111, in that case let's leave the text as it is. Could you please close the two issues you linked from here? Thanks.

**Comment by flip111** [comment]
I was referring to line 24 which you commented on. Since this PR is now closed i opened a new PR with only the pieces that didn't get negative feedback. https://github.com/rust-embedded/book/pull/172 I will close the second issue later

**Pull request closed without merging**